woocommerce/woocommerce-ios
# [Scan to Update Inventory] Inventory Screen UI

Closes: #11307 


## Description

With this PR we implement the UI of the update inventory screen:

- Layout
- Strictly layout logic: when showing the keyboard the first item, we change the button title. Anytime the keyboard is on screen, the image is hidden
- View model:  populate the layout with the inventory item. Here the challenge was to abstract the screen for products and variations, which have a slightly different structure.

## Testing instructions


1. Go to products
2. Scan a product whose inventory can be managed (`manageStock` property to true). The Inventory Update screen should appear.
3. Tap on the quantity, and make sure that the button title changes and the image disappears while the keyboard is onscreen.

Repeat with a product variation. See that the name is updated after a while (we have to retrieve the parent's product name)

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/3e72e4c7-61ba-44ca-9809-9703d5949e55

---
